+adi/+AD4170/Base.m

```diff
@@ -0,0 +1,125 @@
+classdef Base < adi.common.Rx & adi.common.RxTx & ...
+         matlabshared.libiio.base & adi.common.Attribute & ...
+         adi.common.RegisterReadWrite & adi.common.Channel
+    % AD4170/AD4190 Precision ADC Class
+    % AD4190 is 8 channel ADC
+    % AD4170 is a 8 channel ADC with additional filter configurations and other added features
+
+    properties (Nontunable)
+        % Fs for configuring the sampling rate of the device.
+        % It supports a numeric value. Please refer to the DS
+        % for more details on configuring FS.
+        % Fs = 1 corresponds to 500ksps with the sinc5 filter
+        Fs = 1
+
+        %AdcMode ADC Mode
+        % ADC Mode for data capture, represented as a string.
+        % It supports the following: 'Continuous_Conversion',
+        % 'Continuous_Conversion_FIR', 'Continuous_Conversion_IIR',
+        % 'Standby', 'Power_Down', 'Idle'
+        AdcMode = 'Continuous_Conversion'
+
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.
+        SamplesPerFrame = 400
+    end
+    
+    % isOutput
+    properties (Hidden, Nontunable, Access = protected)
+        isOutput = false
+    end
+
+    properties (Nontunable, Hidden, Constant)
+        Type = 'Rx'
+    end
+
+    properties (Nontunable, Hidden)
+        Timeout = Inf
+        kernelBuffersCount = 1
+        dataTypeStr = 'int32'
+        phyDevName
+        devName
+    end
+
+    properties (Hidden, Constant)
+        ComplexData = false
+    end
+
+    methods
+    %% Constructor
+        function obj = Base(phydev,dev,varargin)
+            coder.allowpcode('plain');
+            % Initialize the Rx object
+            obj = obj@matlabshared.libiio.base(varargin{:});
+            obj.enableExplicitPolling = false;
+            obj.EnabledChannels = 1;
+            obj.BufferTypeConversionEnable = true;
+            obj.phyDevName = phydev;
+            obj.devName = dev;
+            obj.uri = 'ip:analog.local';
+        end
+
+        function flush(obj)
+            % Flush the buffer
+            flushBuffers(obj);
+        end
+
+        function delete(obj)
+            % Destructor
+            delete@adi.common.RxTx(obj);
+        end
+
+        function set.AdcMode(obj, value)
+            % Set device ADC Mode
+            obj.AdcMode = value;
+            if obj.ConnectedToDevice
+                obj.setDeviceAttributeRAW('adc_mode', value);
+            end
+        end
+
+        function set.Fs(obj, value)
+            % Set device Fs
+            obj.Fs = value;
+            if obj.ConnectedToDevice
+                obj.setAttributeRAW('voltage0', 'fs', value, false, obj.iioDev);
+                obj.setAttributeRAW('voltage1', 'fs', value, false, obj.iioDev);
+                obj.setAttributeRAW('voltage2', 'fs', value, false, obj.iioDev);
+                obj.setAttributeRAW('voltage3', 'fs', value, false, obj.iioDev);
```

Comment:
nit: Shall we implement in this way in a loop?
https://github.com/analogdevicesinc/PrecisionToolbox/blob/76240ce259bc1b6ef975a405c0ee9f389294113b/%2Badi/%2BAD7124/Base.m#L86

Similar comment for `setupInit()` function